web/packages/teleterm/package.json

```diff
@@ -54,8 +54,8 @@
     "split2": "4.1.0",
     "ts-loader": "^9.3.1",
     "winston": "^3.3.3",
-    "xterm": "^4.15.0",
-    "xterm-addon-fit": "^0.5.0",
+    "xterm": "^5.0.0",
+    "xterm-addon-fit": "^0.7.0",
```

Comment:
Can you remove the carets plz. 

Also can this just be removed and we rely on the package in the root repo?

Reply:
> Can you remove the carets plz.

I see that there was a discussion about it in the other PR, I'm going to move it to Slack as it might be a more suited medium for it.

---

> Also can this just be removed and we rely on the package in the root repo?

I can try this tomorrow, but I'm not sure if it's fine for a package to import another package that's not declared in the deps. I remember running into yarn or some other tool complaining about this but maybe I'm misremembering things.

Reply:
I've looked briefly into adding deps to the root package.json when using workspaces. I found these two threads helpful:

* https://stackoverflow.com/questions/51235261/why-does-yarn-warn-when-adding-a-dependency-to-the-root-workspaces-package-json
* https://www.reddit.com/r/javascript/comments/9t6yht/yarn_workspaces_why_is_adding_something_to_the/

Yarn is fine with adding devDependencies to the root package.json but it will warn you when adding a regular dep:

> Running this command will add the dependency to the workspace root rather than the workspace itself, which might not be what you want - if you really meant it, make it explicit by running this command again with the -W flag (or --ignore-workspace-root-check).

Based on that thread, one bad thing that can happen is if you were to publish any of your packages to the registry, then it'll install fine in your monorepo but will fail if downloaded from the registry as some required packages will be missing from its package.json. I don't think we'll be ever publishing those to a registry as we write applications not libraries.

Someone also mentioned that they worked in a big company where it'd be super hard to upgrade a package defined in the root package.json because it'd mean going through all other packages and making sure that they work fine with the new version. I suppose for us for now it'd be a desired trait as we do want to make sure teleport and teleterm use the same package version.

So while being a bit unorthodox, it doesn't seem like it'd be a problem but then why use workspaces in the first place? I'd probably seek a buy in from the rest of the team though together with some broader strategy on how to approach packages and workspaces in the project rather than moving those deps one by one, so I'm going to keep it as it is for now.